dot-net-sdk/store/ConfigurationStore.cs

```diff
@@ -7,69 +7,66 @@ namespace eppo_sdk.store;
 
 public class ConfigurationStore : IConfigurationStore
 {
-    private readonly MemoryCache _flagConfigurationCache;
-    private readonly MemoryCache _metadataCache;
-    private readonly MemoryCache _banditModelCache;
-    private readonly IConfigurationRequester _requester;
-    private const string BANDIT_FLAGS_KEY = "banditFlags";
-    private const string KEY_FLAG_CONFIG_VERSION = "ufcVersion";
+    private readonly MemoryCache ufcCache;
+    private readonly MemoryCache metadataCache;
+    private readonly MemoryCache banditCache;
 
-    private static readonly object Baton = new();
+    private readonly MemoryCacheEntryOptions cacheOptions = new MemoryCacheEntryOptions().SetSize(1);
+
+    /// <summary>
+    /// Used for concurrent-read and exclusive-write locking.
+    /// </summary>
     private readonly ReaderWriterLockSlim cacheLock = new();
 
-    public ConfigurationStore(IConfigurationRequester requester,
-                              MemoryCache flagConfigurationCache,
+    public ConfigurationStore(MemoryCache flagConfigurationCache,
                               MemoryCache banditModelCache,
                               MemoryCache metadataCache)
     {
-        _requester = requester;
-        _flagConfigurationCache = flagConfigurationCache;
-        _banditModelCache = banditModelCache;
-        _metadataCache = metadataCache;
-    }
-
-    private void SetFlag(string key, Flag flag)
-    {
-        _flagConfigurationCache.Set(key, flag, new MemoryCacheEntryOptions().SetSize(1));
-    }
-
-    private void SetBanditModel(Bandit banditModel)
-    {
-        _banditModelCache.Set(
-            banditModel.BanditKey,
-            banditModel,
-            new MemoryCacheEntryOptions().SetSize(1));
-    }
-
-    private void SetBanditFlags(BanditFlags banditFlags)
-    {
-        _metadataCache.Set(BANDIT_FLAGS_KEY, banditFlags, new MemoryCacheEntryOptions().SetSize(1));
+        this.ufcCache = flagConfigurationCache;
+        this.banditCache = banditModelCache;
+        this.metadataCache = metadataCache;
     }
 
-    public BanditFlags GetBanditFlags()
+    public void SetConfiguration(IEnumerable<Flag> flags, IEnumerable<Bandit> bandits, IDictionary<string, object> metadata)
```

Comment:
This is done so that all of the configuration can be set within the same threadsafe locked block. While it is the responsibility of ConfigurationRequester to determine what config to load and set (flags, banditFlagVariations and Bandits), I wanted to keep all the locking mechanics strictly within `ConfigurationStore`. Setting the flag config, banditFlagVariations and Bandit Models (if applicable) outside of an idempotent write can lead to other threads reading incomplete configuration.